platform-sdk/docs/proposals/consensus-layer/Consensus-Layer.md

```diff
@@ -0,0 +1,720 @@
+back up without incurring the high cost of a reconnect. This may occur during normal operation, if a node is
+experiencing stress, but is particularly likely just after having performed a reconnect. A recently reconnected node may
+be several minutes behind its neighbors, but still be able to catch the rest of the way up by receiving those older
+events through gossip.
+
+#### Neighbor Discipline
+
+If a neighbor misbehaves, the Gossip module will notify the Sheriff module that one of its neighbors is misbehaving. For
+example, if a neighbor is not responding to requests, even after repeated attempts to make a TCP connection with it, it
+may be "bad". Or if the neighbor is sending events that exceed an acceptable rate, or exceed an acceptable size, then it
+is "bad". Or if the events it sends cannot be parsed, or are signed incorrectly, or in other ways fail validation, then
+it is "bad". There may be additional rules by the Gossip module or others (such as Event Intake detecting branching)
+that could lead to a neighbor being marked as "bad". A "bad" node may be dishonest, or it may be broken. The two cases
+may be indistinguishable, so punishment must be adjusted based on the severity of the behavior.
+
+If the Sheriff decides that the neighbor should be penalized, then it will instruct the Gossip module to "shun" that
+neighbor. "Shunning" is a unilateral behavior that one node can take towards another, where it terminates the connection
+and refuses to work further with that neighbor. If the Sheriff decides to welcome a neighbor back into the fold, it can
+instruct the Gossip module to "welcome" the neighbor back.
+
+#### Falling Behind
+
+When a node is operating, it receives events from its neighbors through gossip. If the node for some reason is unable to
+receive and process events fast enough, it may start to "fall behind" the other nodes. Perhaps Bob is processing round
+200 while Alice is still on round 100. If Alice continues to fall farther and farther behind, the time may come when she
+can no longer get old events from her neighbors. From the perspective of the neighbor, the events Alice says she needs
+may have expired, and Bob may no longer be holding those events.
+
+If this happens, then Alice has "fallen behind" and must reconnect. There is no longer any chance that she can get the
+events she needs through gossip alone. Gossip will detect this situation and make a call through the Consensus module
+interface to notify Execution that the node is behind. Execution will then initiate reconnect.
+
+Fundamentally, Execution is responsible for reconnect, but it cannot differentiate between a quiescent network or a
+behind node just by looking at the lack of rounds coming from Consensus. So Consensus **must** tell Execution if it
+knows that it cannot proceed in creating rounds because the node itself is bad, vs. a quiescent network which is
+able to create rounds (if there were any transactions).
```

Comment:
This paragraph was written when the document was proposing that consensus send a stream of rounds to execution in a fire-and-forget way (not execution asking for rounds, but consensus just shooting them out). In this world, you might be tempted to think "I (execution) can tell that I'm behind because I'm not getting any rounds at all, and that only happens if I'm not getting events, and that only happens if I'm so far behind that nobody will send me stuff anymore". But that doesn't work, because quiescence is another situation where events may not be sent and you're not behind.

The point of the paragraph was to say "not so fast! There are other reasons besides falling behind why you may not get rounds very frequently. So don't jump to conclusions!".

It can probably just be removed now or reworded. It wasn't wrong, just maybe not useful.

Reply:
Removing